core/domain/base_model_validators.py

```diff
@@ -84,12 +88,57 @@ def __init__(
             model_class: ClassObject. The external model class.
             model_ids: list(str). The list of external model ids to fetch the
                 external models.
+
+        Raises:
+            Exception. This class was used instead of using the
+                UserSettingsModelFetcherDetails for the UserSettingsModel.
         """
+        if model_class == user_models.UserSettingsModel:
+            raise Exception(
+                'Please use the UserSettingsModelFetcherDetails ' +
+                'for UserSettingsModel')
         self.field_name = field_name
         self.model_class = model_class
         self.model_ids = model_ids
 
 
+class UserSettingsModelFetcherDetails(python_utils.OBJECT):
+    """Value object providing the class and ids to fetch the
+    user settings model.
+    """
+
+    def __init__(
+            self, field_name, model_ids,
+            system_user_ids_removed=False, pseudonymous_ids_removed=False):
+        """Initializes the UserSettingsModelFetcherDetails domain object.
+
+        Args:
+            field_name: str. A specific name used as an identifier by the
+                storage model which is used to identify the external model
+                reference. For example: 'exp_ids': ExplorationModel, exp_ids
+                is the field name to identify the external model
+                ExplorationModel.
+            model_ids: list(str). The list of external model ids to fetch the
+                external models.
+            system_user_ids_removed: bool. Whether to remove the system user IDs
+                from the model_ids.
```

Comment:
...from the model_ids before attempting to fetch the corresponding models.

(Ditto below for pseudonymous_ids_removed)

Reply:
Done.